### PR TITLE
Feat: Docs Detangling Tooling

### PR DESCRIPTION
**Pull request by rsylvian**
# Description
- Add custom Jekyll plugins to generate JSON data that can be used by `docs-platform`
- Remove `src-shared` submodule because it is handled by `docs-platform` directly now 
- Adapt CI config to make sure it still builds correctly
- Add @teesloane's work on API building
- Cleaned up various small pieces that are not needed anymore (JS experiments, JS UI)

# Reasons
This is the first step in getting some of that Docs Detangling work to `master`. Once this is merged, this will allow us to run `docs-platform` directly from `master` instead of the `rs/feat/docs-detangling` branch which will make the content publishing workflow much more streamlined.

### CODE REVIEW

File: jekyll/_data/sidenav.yml [hunk 1]
```diff
@@ -1167,13 +1167,16 @@ ja:
...
+    - name: Languages
+      i18n_name: 言語
+      children:
+        - name: Overview
+          link: ja/2.0/examples-and-guides-overview/
+          i18n_name: 概要
+        - name: Node
+          link: ja/2.0/language-javascript/
+        - name: Python
+          link: ja/2.0/language-python/
```
Comment (rsylvian) [inline jekyll/_data/sidenav.yml:1179]: For Docs team: this is the reason why you are tagged on that PR. I changed the JA sidebar data to make sure it doesn't introduce errors. I did the exact same work (but for EN) last week: https://github.com/circleci/circleci-docs/pull/6970/files

**Review by Mayuko-Sa — APPROVED** [review]

File: jekyll/_data/sidenav.yml (hunk 1 above)
Comment (Mayuko-Sa) [reply jekyll/_data/sidenav.yml:1179]: Hey @rsylvian thank you for taking care of JA sidebar 😸 Approved!

**Review by chantelle11 — APPROVED** [review]
Great work on this!

**Review by tshen815 — APPROVED** [review]
Thanks for the breakdown on this 🙏

**Pull request merged**